src/items.rs

```diff
@@ -1289,11 +1312,11 @@ pub(crate) fn format_struct_struct(
             context,
             g,
             context.config.brace_style(),
-            if fields.is_empty() {
-                BracePos::ForceSameLine
-            } else {
-                BracePos::Auto
-            },
+            set_brace_pos(
+                !fields.is_empty(),
+                span.hi() <= body_lo + BytePos(5),
```

Comment:
To be honest, this check `span.hi() <= body_lo + BytePos(5)` has got me scratching my head and it's not obvious from reading the code what it's supposed to be doing. I'm all for moving this logic into  a standalone function, but I think this check could be simplified to something like:

```rust
            if fields.is_empty() {
                let snippet = context.snippet(mk_sp(body_lo, span.hi()));
                let body_contains_commets = contains_comment(snippet);
                if context.config.version() == Version::Two && body_contains_commets {
                    BracePos::Auto
                } else {
                    BracePos::ForceSameLine
                }
            } else {
                BracePos::Auto
            }
```